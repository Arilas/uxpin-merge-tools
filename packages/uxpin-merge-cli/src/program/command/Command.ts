export enum Command {
  DUMP = 'dump',
  EXPERIMENT = 'experiment',
  INIT = 'init',
  CREATE_APP = 'create-app',
  GENERATE_PRESETS = 'generate-presets',
  PUSH = 'push',
  SERVER = 'server',
  SUMMARY = 'summary',
}

export const DEFAULT_COMMAND:Command = Command.EXPERIMENT;
