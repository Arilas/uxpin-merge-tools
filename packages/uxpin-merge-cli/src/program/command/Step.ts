import { DSMetadata } from '../DSMeta';

export interface Step {
  shouldRun:boolean;
  exec:(infos:DSMetadata) => any;
}
