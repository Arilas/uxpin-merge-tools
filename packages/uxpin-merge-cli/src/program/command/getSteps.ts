import { BuildOptions } from '../../steps/building/BuildOptions';
import { ProgramArgs, PushProgramArgs, ServerProgramArgs } from '../ProgramArgs';
import { getDumpCommandSteps } from './dump/getDumpCommandSteps';
import { getPushCommandSteps } from './push/getPushCommandSteps';
import { getServerCommandSteps } from './server/getServerCommandSteps';
import { Step } from './Step';
import { getSummaryCommandSteps } from './summary/getSummaryCommandSteps';

export function getSteps(args:ProgramArgs):Step[] {
  if (args.command === 'server') {
    return getServerCommandSteps(getBuildOptions(args), args);
  }

  if (args.command === 'push') {
    return getPushCommandSteps(getBuildOptions(args));
  }

  if (args.command === 'dump') {
    return getDumpCommandSteps();
  }

  if (args.command === 'summary') {
    return getSummaryCommandSteps();
  }

  return [];
}

function getBuildOptions(args:ServerProgramArgs | PushProgramArgs):BuildOptions {
  const { cwd, webpackConfig, wrapper } = args;
  return {
    projectRoot: cwd,
    webpackConfigPath: webpackConfig,
    wrapperPath: wrapper,
  };
}
