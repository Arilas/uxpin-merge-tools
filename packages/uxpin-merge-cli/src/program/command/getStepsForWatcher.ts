import { ProgramArgs } from '../args/ProgramArgs';
import { getExperimentationWatchCommandSteps } from './experimentation/getExperimentationWatchCommandSteps';
import { Step } from './Step';

export function getStepsForWatcher(args:ProgramArgs):Step[] {
  if (args.command === 'experiment') {
    return getExperimentationWatchCommandSteps(args);
  }

  return [];
}
