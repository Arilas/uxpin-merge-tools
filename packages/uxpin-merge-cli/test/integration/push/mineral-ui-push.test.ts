import { runUXPinMergeCommand } from '../../utils/command/runUXPinMergeCommand';
import { setTimeoutBeforeAll } from '../../utils/command/setTimeoutBeforeAll';

const CURRENT_TIMEOUT:number = 75000;
setTimeoutBeforeAll(CURRENT_TIMEOUT);

describe('Pushing mineral-ui design system', () => {
  describe('with required user webpack config', () => {
    let consoleOutput:string;

    beforeAll(async () => {
      const params:string[] = [
        'push',
        '--webpack-config "./webpack.config.js"',
        '--wrapper "./src/library/themes/UXPinWrapper.js"',
      ];

      consoleOutput = await runUXPinMergeCommand({ cwd: 'resources/repos/mineral-ui', params });
    });

    it('prints warnings without stack traces to the console', () => {
      // then
      expect(consoleOutput).toMatchSnapshot();
    });
  });

  describe('without required user webpack config', () => {
    it('throws an error', async () => {
      await expect(runUXPinMergeCommand({ cwd: 'resources/repos/mineral-ui', params: ['push'] }))
        .rejects.toMatch('Module parse failed: Unexpected token');
    });
  });
});
