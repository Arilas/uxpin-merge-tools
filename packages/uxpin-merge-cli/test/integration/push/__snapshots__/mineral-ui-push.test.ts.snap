// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pushing mineral-ui design system with required user webpack config prints warnings without stack traces to the console 1`] = `
"warning Cannot compute default value for property \`triggerTitle\`

SyntaxError: Unexpected token :

 in:
src/library/Card/CardFooter/CardFooter.js

[32mâœ… Library is up-to-date!
[0m"
`;
